### PR TITLE
chore: better debug info + a clear check on env variable DISABLE_DSC_CONFIG

### PR DESCRIPTION
**Pull request by zdtsw**
hard to know what user is doing in the cluster if they are running for Managed cluster but no DSCI CR gets auto created. 
just a bit more info to help debug

## Description

thread: https://redhat-internal.slack.com/archives/C064Z5S84F3/p1713957713956549?thread_ts=1713950565.201169&cid=C064Z5S84F3

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -207,8 +207,10 @@ func main() { //nolint:funlen
 		os.Exit(1)
 	}
 	// Check if user opted for disabling DSC configuration
-	_, disableDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
-	if !disableDSCConfig {
+	disableDSCConfig, existDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
+	if existDSCConfig {
+		setupLog.Info("DSCI auto creation is disabled")
+	} else if disableDSCConfig == "false" || !existDSCConfig {
```
Comment (ykaliuta) [inline main.go:213]: if we reach this branch existDSCConfig is false
Is the logic
```
	disableVar, varSet := os.LookupEnv("DISABLE_DSC_CONFIG")
	disableDSCConfig := varSet && disableVar != "false"
	if disableDSCConfig {
		setupLog.Info("DSCI auto creation is disabled")
	} else {
		...
```
?

Comment (zdtsw) [reply main.go:213]:
```
	disableVar, varSet := os.LookupEnv("DISABLE_DSC_CONFIG")
	disableDSCConfig := varSet || disableVar != "false"
	if disableDSCConfig {
		setupLog.Info("DSCI auto creation is disabled")
	} else {
		...
``` 
i think it is || not && in your example

Comment (zdtsw) [reply main.go:213]: if not set variable => create DSCI
if set variable + no value => skip create
if set variable to true => skip create
if set variable to false => create DSCI

```
disableDSCConfig, existDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
if !existDSCConfig || disableDSCConfig == "false" {
// create
} else {
 setupLog.Info("DSCI auto creation is disabled")
}


```

Comment (ykaliuta) [reply main.go]: > if not set variable => create DSCI if set variable + no value => skip create if set variable to true => skip create if set variable to false => create DSCI
> 
> ```
> disableDSCConfig, existDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
> if !existDSCConfig || disableDSCConfig == "false" {
> // create
> } else {
>  setupLog.Info("DSCI auto creation is disabled")
> }
> ```

This is actually the same as in my snippet,  https://en.wikipedia.org/wiki/De_Morgan%27s_laws

Comment (ykaliuta) [reply main.go]: I personally buy Allen I. Holub's argumentation to put shortest branch first like in my snippet, but will not insist.

Comment (zdtsw) [reply main.go]: sorry, was fixing another PR. 
updated now

Comment (VaishnaviHire) [reply main.go]: @zdtsw Note We would need to update documentation in downstream to ensure we do not break existing installations

Comment (zdtsw) [reply main.go]: i will update README for this part, but i dont think it was officially documented in downstream for user to use this env variable. i can check with "docs"

Comment (zdtsw) [reply main.go]: i see https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/README.md?plain=1#L72 is already there, just we did not implemented the "false" but only check if the env variable has been set or not

**Pull request merged**